sdk/samples/iot/templates/iothub/test-resources-post.ps1

```diff
@@ -30,7 +30,7 @@ $iothubName = $DeploymentOutputs['IOT_HUB_NAME']
 ###### X509 setup ######
 # Generate certificate
 openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
-openssl req -new -days 12 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$deviceID"
+openssl req -new -days 12 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -extensions client_auth -config x509_config.cfg -subj "/CN=$deviceID"
```

Comment:
are these `-extensions client_auth` needed even though they're also in the config?

Reply:
On my Ubuntu 20.04 (tested on 2 systems), without that addition the cert generated is completely ignoring the config. Generates a V1 cert:

```
 Data:
        Version: 1 (0x0)
        Serial Number:
            04:41:c1:db:6d:a4:a5:0d:2c:3e:ec:9b:ca:b4:5d:c8:bd:02:de:bf
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN = dev1-ecc
        Validity
            Not Before: Nov 11 02:36:42 2021 GMT
            Not After : Nov 23 02:36:42 2021 GMT
        Subject: CN = dev1-ecc
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:af:a5:3e:62:0d:ce:6d:c3:46:82:8b:36:5f:5e:
                    dd:70:81:9d:d6:44:c9:a3:ff:fa:44:c9:ea:7e:2b:
                    58:cc:36:ee:6d:c0:7e:b6:4f:c1:c6:e5:f1:4c:05:
                    72:9d:7a:c0:e9:42:a8:21:3d:da:f6:60:76:55:b3:
                    50:97:01:8f:c9
                ASN1 OID: prime256v1
                NIST CURVE: P-256
    Signature Algorithm: ecdsa-with-SHA256
         30:46:02:21:00:ea:b2:37:b8:a0:5b:3a:26:80:7b:00:af:1f:
         86:e4:70:a5:40:ce:c7:61:c7:16:ba:59:9e:23:15:4b:44:de:
         22:02:21:00:da:e9:c0:a1:50:bf:cb:ae:15:42:97:61:7d:ac:
         2b:fd:62:bd:82:cd:01:c2:4c:07:80:fb:2b:61:3c:03:f6:50
SHA1 Fingerprint=C1:D2:85:3E:8A:BF:78:C7:47:A5:26:6B:F6:90:ED:43:58:F8:2B:49
```


With the changes I get the correct V3 cert:

```
    Data:
        Version: 3 (0x2)
        Serial Number:
            0a:a8:ca:f6:df:98:f7:ac:b1:8c:c0:45:d0:9c:c6:54:bf:22:cb:e0
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN = dev1-ecc
        Validity
            Not Before: Nov 11 02:37:18 2021 GMT
            Not After : Nov 23 02:37:18 2021 GMT
        Subject: CN = dev1-ecc
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:af:a5:3e:62:0d:ce:6d:c3:46:82:8b:36:5f:5e:
                    dd:70:81:9d:d6:44:c9:a3:ff:fa:44:c9:ea:7e:2b:
                    58:cc:36:ee:6d:c0:7e:b6:4f:c1:c6:e5:f1:4c:05:
                    72:9d:7a:c0:e9:42:a8:21:3d:da:f6:60:76:55:b3:
                    50:97:01:8f:c9
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Key Usage: 
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Client Authentication
    Signature Algorithm: ecdsa-with-SHA256
         30:46:02:21:00:b0:e0:5a:ce:d0:53:c8:2d:82:5e:f0:36:fa:
         8e:20:ca:80:22:c4:6e:5b:5d:80:64:79:92:c7:ee:72:0f:08:
         0f:02:21:00:c5:3e:d3:81:63:2b:6f:6b:24:60:f8:51:d5:57:
         ac:48:03:19:fd:13:73:11:1b:90:4c:eb:e4:9b:3d:20:fc:ea
SHA1 Fingerprint=DE:30:18:18:2F:90:D6:C7:AA:6F:AC:31:93:59:97:C3:77:FE:02:54
```

Worth giving it a try on your system to check if this is not something OpenSSL dependent that I've overlooked.